### ISSUE TITLE
AWS Lambda image is invalid when pushed with `oci_push`

### ISSUE THREAD

**Issue by nnmm** [bazel-contrib/rules_oci#777]
I have built an image for AWS lambda in `rules_oci`, but when deploying that image I'm getting an error that says

```
InvalidImage: ImageLayerFailure: UnsupportedImageLayerDetected - Layer Digest sha256:eb0495635521a07c170768de6c1b935c89ffd8e9954fe9bb0d6bea58bdb62777
```

That layer is the first layer that `rules_oci` adds to the base image, and it contains just a single file wrapped into a `pkg_tar` (from `rules_pkg`). I am pushing it to Amazon ECR with `oci_push`.

When I load the image into Docker with `oci_load` and then push with `docker push`, it works, and I can see that this image is only 193.59 MB in ECR as opposed to the 263.42 MB for the `oci_push`ed image. Why is that?

<details><summary>Previous, probably irrelevant idea</summary>
<p>

The only idea I have right now what could cause that error is that the [docs](https://docs.aws.amazon.com/lambda/latest/dg/python-image.html#python-image-instructions) for building an AWS lambda layer say to add the `--provenance=false` argument to `docker build` - is there an equivalent for this flag in `rules_oci`, or am I correct to assume that provenance attestation is off by default and only enabled by the new cosign rules?

</p>
</details>

**Comment by nnmm** [comment]
When I pull both images from ECR with Docker, they show up with the same image ID and size (629MB) in `docker image ls`.

**Comment by plobsing** [comment]
> When I load the image into Docker with oci_load and then push with docker push, it works, and I can see that this image is only 193.59 MB in ECR as opposed to the 263.42 MB for the oci_pushed image. Why is that?

Most likely the [media type](https://github.com/opencontainers/image-spec/blob/main/layer.md) of your layer is different. By default `pkg_tar` gives you a tarball (media type: `application/vnd.oci.image.layer.v1.tar`); most image build tools default to using gzipped tarballs (media type: `application/vnd.oci.image.layer.v1.tar+gzip`). Both are supposed to be part of the spec, but it is possible Lambda only honours the latter since the former are less common. You can get a gzipped tarball from `pkg_tar` by specifying `compression = "gz"`.

**Comment by nnmm** [comment]
Thank you very much @plobsing, that was the solution! I added `extension = "tar.gz"` to every `pkg_tar` layer and it works now.
